### PR TITLE
Bugfixes in multiple areas

### PR DESCRIPTION
**Pull request by theplasmastorm**
- Added support for filtering on boolean keys
- Fixed a bug in filtering via datestamps and timestamps because the formatting was incorrect
- Fixed missing sample data in order table and order items table
- Changes to the unit tests' upgrade tests to account to explicitly name properties
- Added a structure called "DifferentPK", which tests weird and unusual key types inside HarmonyCore
- Fixed bugs in GenerateTestValues where values would generate incorrectly based on assumption of primary key types

@hippiehunter may be easier to review via commits rather than giant file change set

### CODE REVIEW

File: Services.Test/UnitTests/HandCrafted.dbl [hunk 1]
```diff
@@ -1616,6 +1616,26 @@ namespace Services.Test.UnitTests
...
 
+        class DifferentPKWithCount
+            {JsonProperty("@odata.count")}
+            public readwrite property Count, long
+            {JsonProperty("value")}
+            public readwrite property Value, [#]@Services.Models.Differentpk
+        endclass
+
+        {TestMethod}
+        public method DifferentPKJoin, void
```
Comment (hippiehunter) [inline Services.Test/UnitTests/HandCrafted.dbl:1627]: I think we should have a bit more of this. as an example see GetCustomerGeLePhoneRef

File: Templates/GenerateTestValues.tpl [hunk 2]
```diff
@@ -102,15 +100,15 @@ namespace Services.Test.GenerateValues
 ;//
   <ALTERNATE_KEY_LOOP_UNIQUE>
     <SEGMENT_LOOP>
-            TestConstants.Instance.Get<StructureNoplural>_ByAltKey_<KeyName>_<SegmentName> = <IF DATEORTIME>DateTime.ParseExact((string)(<structureNoplural>.<segment_name>), "yyyyMMdd", System.Globalization.CultureInfo.InvariantCulture)<ELSE><IF SEG_TYPE_FIELD><structureNoplural>.<segment_name><ELSE SEG_TYPE_LITERAL>"<SEGMENT_LITVAL>"</IF SEG_TYPE_FIELD></IF DATEORTIME>
+            TestConstants.Instance.Get<StructureNoplural>_ByAltKey_<KeyName>_<SegmentName> = <IF DATEORTIME>DateTime.ParseExact((string)(<structureNoplural>.<segment_name>), <IF DATE>"yyyyMMdd"<ELSE>"hhmmss"</IF DATE>, System.Globalization.CultureInfo.InvariantCulture)<ELSE><IF SEG_TYPE_FIELD><structureNoplural>.<segment_name><ELSE SEG_TYPE_LITERAL>"<SEGMENT_LITVAL>"</IF SEG_TYPE_FIELD></IF DATEORTIME>
```
Comment (hippiehunter) [inline Templates/GenerateTestValues.tpl:103]: There's a ton of date and time format strings, I don't think we can get away with hard coding just 1 here. see date conversion routine here https://github.com/Synergex/HarmonyCore/blob/d2cf0978d156b831912d8c37224e99fb99903a65/HarmonyCore/Converters/CustomDateRoutines.dbl#L117

File: Templates/ODataUnitTests.tpl [hunk 3]
```diff
@@ -220,7 +220,7 @@ namespace <NAMESPACE>
       <IF DEFINED_ENABLE_AUTHENTICATION>
             client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer",UnitTestEnvironment.AccessToken)
       </IF DEFINED_ENABLE_AUTHENTICATION>
-            data request = String.Format("/odata/v<API_VERSION>/<StructurePlural>(<SEGMENT_LOOP><SegmentName>=<IF ALPHA>'</IF ALPHA>{<SEGMENT_NUMBER>}<IF ALPHA>'</IF ALPHA><,></SEGMENT_LOOP>)", "", <SEGMENT_LOOP>TestConstants.Instance.Get<StructureNoplural>_ByAltKey_<KeyName>_<SegmentName><IF DATE>.ToString("yyyy-MM-dd")</IF DATE><,></SEGMENT_LOOP>)
+            data request = String.Format("/odata/v<API_VERSION>/<StructurePlural>(<SEGMENT_LOOP><SegmentName>=<IF ALPHA>'</IF ALPHA>{<SEGMENT_NUMBER>}<IF ALPHA>'</IF ALPHA><,></SEGMENT_LOOP>)", "", <SEGMENT_LOOP>TestConstants.Instance.Get<StructureNoplural>_ByAltKey_<KeyName>_<SegmentName><IF DATE>.ToString("yyyy-MM-dd")</IF DATE><IF TIME_HHMM>.ToString("hh:mm")</IF TIME_HHMM><IF TIME_HHMMSS>.ToString("hh:mm:ss")</IF TIME_HHMMSS><IF BOOLEAN>.ToString().ToLower()</IF BOOLEAN><,></SEGMENT_LOOP>)
```
Comment (hippiehunter) [inline Templates/ODataUnitTests.tpl:223]: Same comment as above regarding date formats

File: Services.Test/UnitTests/HandCrafted.dbl (hunk 1 above)
Comment (theplasmastorm) [reply Services.Test/UnitTests/HandCrafted.dbl:1627]: Added two more tests.
See c032eebba73e134108efa6f12679b870588d8859

File: Templates/GenerateTestValues.tpl (hunk 2 above)
Comment (theplasmastorm) [reply Templates/GenerateTestValues.tpl]: Fixed with commit f21ca12

File: Templates/ODataUnitTests.tpl (hunk 3 above)
Comment (theplasmastorm) [reply Templates/ODataUnitTests.tpl:223]: No necessary changes here as it is a Datetime type

**Review by hippiehunter — APPROVED** [review]

**Pull request merged**